### PR TITLE
Fix jar path and add missing ones for spark jobs

### PR DESCRIPTION
**Pull request by aashudwivedi**
Fix path of jars / add missing jars in spark job

remove print, reduce clutter

## Description ##
Path of jar files in tests for MXNet on Spark are incorrect. This PR fixes it.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala [hunk 1]
```diff
@@ -81,17 +81,15 @@ trait SharedSparkContext extends FunSuite with BeforeAndAfterEach with BeforeAnd
   }
 
   private def getJarFilePath(root: String): String = {
-    for (platform <- List("linux-x86_64-cpu", "linux-x86_64-gpu", "osx-x86_64-cpu")) {
```
Comment (ChaiBapchya) [inline scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala]: by removing this for loop; does it mean the path is independent of platform?

Comment (piyushghai) [reply scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala]: Yes. The path for the jars is platform agnostic now.

**Comment by vandanavk** [comment]
@mxnet-label-bot add [pr-awaiting-review, Scala]

@aashudwivedi could you re-trigger the CI?

**Comment by aashudwivedi** [comment]
I've re-triggered the CI, all tests have passed now. 
Thanks for having a look.

File: scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala [hunk 2]
```diff
@@ -81,19 +81,18 @@ trait SharedSparkContext extends FunSuite with BeforeAndAfterEach with BeforeAnd
...
-            !pathname.getAbsolutePath.contains("javadoc") &&
-            !pathname.getAbsolutePath.contains("sources")
-        }
-      })
-      if (jarFiles != null && jarFiles.nonEmpty) {
-        return jarFiles.head.getAbsolutePath
+    val jarFiles = new File(s"$root/target/").listFiles(new FileFilter {
+      override def accept(pathname: File) = {
+        pathname.getAbsolutePath.endsWith(".jar") &&
+          !pathname.getAbsolutePath.contains("bundle") &&
```
[the remark this answers is not in the document]
Comment (piyushghai) [reply scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala]: I agree with Zach here. We should also ignore the others

File: scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala [hunk 3]
```diff
@@ -80,14 +80,18 @@ trait SharedSparkContext extends FunSuite with BeforeAndAfterEach with BeforeAnd
...
-  private def getJarFilePath(root: String): String = {
-    val jarFiles = new File(s"$root/target/").listFiles(new FileFilter {
+  private def findJars(root: String): Array[File] = {
+    val excludedSuffixes = List("bundle", "src", "javadoc", "sources")
+    new File(root).listFiles(new FileFilter {
       override def accept(pathname: File) = {
         pathname.getAbsolutePath.endsWith(".jar") &&
-          !pathname.getAbsolutePath.contains("bundle") &&
-          !pathname.getAbsolutePath.contains("src")
+          excludedSuffixes.map(!pathname.getAbsolutePath.contains(_)).forall(identity)
```
Comment (zachgk) [inline scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala]: You can just use excludedSuffixes.forall(!pathname.getAbsolutePath.contains(_))

**Comment by aashudwivedi** [comment]
> Looks okay to me!
> Thanks for fixing this and welcome to the MXNet community :)
Thanks you :)

File: scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala (hunk 3 above)
Comment (aashudwivedi) [reply scala-package/spark/src/test/scala/org/apache/mxnet/spark/SharedSparkContext.scala]: Done.

**Comment by ankkhedia** [comment]
@lanking520 ping for review/merge.

**Pull request merged**